tests/Unit/ServiceBasedMerchant/ServiceBasedMerchantStateTest.php

```diff
@@ -0,0 +1,762 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\ServiceBasedMerchant;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\ServiceBasedMerchant\ServiceBasedMerchantState;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\ContainerAwareUnitTest;
+use PHPUnit\Framework\MockObject\MockObject;
+use WC_Helper_Product;
+use WC_Product;
+
+/**
+ * Class ServiceBasedMerchantStateTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\ServiceBasedMerchant
+ */
+class ServiceBasedMerchantStateTest extends ContainerAwareUnitTest {
```

Comment:
@jamesmorrison once we've simplified the implementation for the `ServiceBasedMerchantState` lets make sure to update tests here also.